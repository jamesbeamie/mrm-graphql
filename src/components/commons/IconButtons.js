import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { deleteIcon, editIcon } from '../../utils/images/images';

class IconButtons extends Component {
  static propTypes = {
    buttonText: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    modalButtonClassName: PropTypes.string,
    openModal: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    btnImage: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  };

  static defaultProps = {
    buttonText: '',
    modalButtonClassName: 'button',
    disabled: false,
    btnImage: '',
  };

  /**
   * Fires an event when the text on a reusable button changes
   *
   * @param {string} text
   *
   * @returns {string}
   */
  onButtonTextChange = (text) => {
    let buttonText;
    if (text === 'Delete') {
      buttonText = <img src={deleteIcon} alt="Delete" />;
    } else if (text === 'Edit') {
      buttonText = <img src={editIcon} alt="Edit" />;
    } else {
      buttonText = text;
    }
    return buttonText;
  };
  render() {
    const {
      buttonText, modalButtonClassName, disabled, btnImage,
    } = this.props;
    return (
      <button
        id="modal-button"
        className={modalButtonClassName}
        onClick={this.props.openModal}
        disabled={disabled}
      >
        {btnImage || this.onButtonTextChange(buttonText)}
      </button>
    );
  }
}

export default IconButtons;
