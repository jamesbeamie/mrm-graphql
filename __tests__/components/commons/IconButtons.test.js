import React from 'react';
import { mount } from 'enzyme';
import IconButtons from '../../../src/components/commons/IconButtons';

describe('IconButtons Component', () => {
  const wrapper = mount(<IconButtons
    buttonText="button"
    modalButtonClassName="button"
    openModal={jest.fn()}
  />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handles onButtonTextChange()', () => {
    expect(wrapper.instance().onButtonTextChange());
  });
});
