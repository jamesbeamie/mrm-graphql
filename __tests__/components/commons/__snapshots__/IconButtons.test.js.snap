// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtons Component renders properly 1`] = `
<IconButtons
  btnImage=""
  buttonText="button"
  disabled={false}
  modalButtonClassName="button"
  openModal={[MockFunction]}
>
  <button
    className="button"
    disabled={false}
    id="modal-button"
    onClick={[MockFunction]}
  >
    button
  </button>
</IconButtons>
`;
